### PR TITLE
feat(Configuration): Add a "host_down_disable_service_checks" option

### PR DESCRIPTION
**Pull request by dmyios**
## Description

- Added a "host_down_disable_service_checks" option to engine configuration.
- Fixed SQL query compatibility issue for duplicating pollers

**Fixes** # MON-125658

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [ ] 24.04.x
- [x] master

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/www/include/configuration/configServers/DB-Func.php [hunk 1]
```diff
@@ -73,11 +73,14 @@ function getAvailableSuffixIds(
...
-        . "ORDER BY suffix";
-    $results = $pearDB->query($query);
+    $serverName = $pearDB->escapeString($serverName);
+    $separator = $pearDB->escapeString($separator);
+    $query = <<<SQL
+        SELECT CAST(SUBSTRING_INDEX(name, '_', -1) AS SIGNED) AS suffix
+        FROM nagios_server WHERE name REGEXP '^"{$serverName}{$separator}[0-9]+$'
+        ORDER BY suffix
+        SQL;
+    $results = $pearDB->executeQuery($query);
```
Comment (callapa) [inline centreon/www/include/configuration/configServers/DB-Func.php:83]: Please use a prepare request.

File: centreon/www/install/php/Update-24.10.0.php [hunk 2]
```diff
@@ -68,7 +68,22 @@
     }
 };
 
+$addDisableServiceCheckColumn = function (CentreonDB $pearDB) use (&$errorMessage): void {
+    $errorMessage = 'Unable to add column host_down_disable_service_checks to table cfg_nagios';
+    if (! $pearDB->isColumnExist('cfg_nagios', 'host_down_disable_service_checks')) {
+        $pearDB->executeQuery(
+            <<<'SQL'
+                ALTER TABLE `cfg_nagios`
+                ADD COLUMN `host_down_disable_service_checks` ENUM('0', '1') DEFAULT '0'
```
Comment (callapa) [inline centreon/www/install/php/Update-24.10.0.php:77]:
```suggestion
                ADD COLUMN `host_down_disable_service_checks` ENUM('0', '1') DEFAULT NULL
```

**Review by callapa — CHANGES_REQUESTED** [review]

File: centreon/www/install/php/Update-24.10.0.php (hunk 2 above)
Comment (dmyios) [reply centreon/www/install/php/Update-24.10.0.php:77]: corrected in installation script

File: centreon/www/include/configuration/configServers/DB-Func.php (hunk 1 above)
Comment (dmyios) [reply centreon/www/include/configuration/configServers/DB-Func.php]: Done

**Review by a-launois — DISMISSED** [review]

File: centreon/www/include/configuration/configServers/DB-Func.php [hunk 3]
```diff
@@ -73,15 +73,20 @@ function getAvailableSuffixIds(
     $serverName = preg_quote($serverName);
 
     // Get list of suffix already used
-    $serverName = CentreonDB::escape($serverName);
-    $query = "SELECT CAST(SUBSTRING_INDEX(name,'_',-1) AS INT) AS suffix "
-        . "FROM nagios_server WHERE name REGEXP '^" . $serverName . $separator . "[0-9]+$' "
-        . "ORDER BY suffix";
-    $results = $pearDB->query($query);
+    $serverName = $pearDB->escapeString($serverName);
+    $separator = $pearDB->escapeString($separator);
```
Comment (dmyios) [inline centreon/www/include/configuration/configServers/DB-Func.php:77]:
```suggestion
```

**Review by callapa — APPROVED** [review]

**Review by a-launois — APPROVED** [review]

**Pull request merged**